### PR TITLE
[MainMenu] Add clear button and icon for search one.

### PR DESCRIPTION
**Pull request by Andrey2470T**
This PR allows to purge the user's search results immediately like as in the creative inventory by just pressing the new 'Clear' button. Also, this adds an icon for the current search button, instead the text. It modifies 'Online' tab and the package manager.

_'Online' tab:_
Screenshot_20200902_235604

_Package Manager:_
Screenshot_20200902_235639

## To do
This PR is Ready for Review.

### CODE REVIEW

**Comment by paramat** [comment]
I assume the 'clear' and 'search' icons are taken from MTG creative mod?
If so, those textures need adding to the texture credits here
https://github.com/minetest/minetest/blob/4ba5046308d6bdf7b38394770c6f82b6927393f2/LICENSE.txt#L22-L25

**Comment by Andrey2470T** [comment]
May anybody please review quickly prior to merging? It doesn't take much time surely as this PR is 
pretty minor.

**Comment by numberZero** [comment]
I’d prefer search-erase-refresh order, more like the creative inventory. Tooltips would be nice too.

**Comment by Andrey2470T** [comment]
I think the clear button yet should be first as this is common for the most of graphic interfaces. I'm unsure about tooltips, I think their icons are already intuitively understandable what their buttons implement.

File: builtin/mainmenu/dlg_contentstore.lua
Comment (sfan5) [inline builtin/mainmenu/dlg_contentstore.lua:343]: whitespace

File: builtin/mainmenu/dlg_contentstore.lua
Comment (sfan5) [inline builtin/mainmenu/dlg_contentstore.lua:356]: indentation is wrong here and in many other places

Comment (Andrey2470T) [reply builtin/mainmenu/dlg_contentstore.lua]: I have fixed all those, but I don't understand how they appeared here. In my original version it looked like properly.

File: builtin/mainmenu/dlg_contentstore.lua
Comment (Andrey2470T) [reply builtin/mainmenu/dlg_contentstore.lua:343]: Do you suggest to add a whitespace in the empty line? What for?

**Comment by Andrey2470T** [comment]
The PR is ready and already reviewed by one core dev. So, merge?

**Comment by Andrey2470T** [comment]
What is a current status of my PR? It is very easy and hanging out in an undefinitive state already two months. No feedback from the core devs up-to-day. Is it approved? If yes, may it be merged now?

File: builtin/mainmenu/dlg_contentstore.lua
Comment (numberZero) [reply builtin/mainmenu/dlg_contentstore.lua:343]: the suggestion is to *remove* whitespace from that line :p

File: builtin/mainmenu/dlg_contentstore.lua
Comment (numberZero) [reply builtin/mainmenu/dlg_contentstore.lua]: > but I don't understand how they appeared here. In my original version it looked like properly.

Of course you don’t. Your editor does. It uses spaces instead of tabs (and there are still some after your fixes). That’s a silly but very popular misfeature. Always disable it when working on Minetest code.

File: builtin/mainmenu/dlg_contentstore.lua
Comment (Andrey2470T) [reply builtin/mainmenu/dlg_contentstore.lua]: Done.

File: builtin/mainmenu/dlg_contentstore.lua
Comment (sfan5) [inline builtin/mainmenu/dlg_contentstore.lua:355]: mixed indentation

this line begins with 12 spaces and 2 tabs, it should be 4 tabs

File: builtin/mainmenu/dlg_contentstore.lua
Comment (sfan5) [inline builtin/mainmenu/dlg_contentstore.lua:512]: same with all of these lines

File: builtin/mainmenu/tab_online.lua
Comment (sfan5) [inline builtin/mainmenu/tab_online.lua:37]: and two times in this file too

**Comment by sfan5** [comment]
The broken indentation could be seen in the lua_lint run too: https://github.com/minetest/minetest/pull/10363/checks?check_run_id=1474999370#step:5:71
You didn't have to wait for me to tell you.

**Comment by Andrey2470T** [comment]
@sfan5: I`ve fixed all what was necessary: added actual tabs in my editor and removed the whitespaces. May it be merged now?

**Comment by Andrey2470T** [comment]
Search-clear-refresh order how was requested by sfan5 on the irc:
Screenshot_20201210_212407
Screenshot_20201210_212419

**Review by sfan5 — APPROVED** [review]
Tested.

**Review by SmallJoker — APPROVED** [review]
Works

**Comment by Andrey2470T** [comment]
Thanks. :)

**Pull request merged**
